java/src/apps/util/issuereporter/swing/IssueReporter.java

```diff
@@ -0,0 +1,388 @@
+        pack();
+    }
+
+    @Override
+    public void insertUpdate(DocumentEvent e) {
+        changedUpdate(e);
+    }
+
+    @Override
+    public void removeUpdate(DocumentEvent e) {
+        changedUpdate(e);
+    }
+
+    @Override
+    public void changedUpdate(DocumentEvent e) {
+        submitBtn.setEnabled(!bodyTA.getText().isEmpty() && !titleText.getText().isEmpty());
+    }
+
+    private void typeCBActionListener(ActionEvent e) {
+        switch (typeCB.getSelectedIndex()) {
+            case BUG:
+                descriptionLabel.setText(Bundle.getMessage("IssueReporter.descriptionLabel.bug"));
+                instructionsLabel.setText(Bundle.getMessage("IssueReporter.instructionsLabel.bug"));
+                if (!typeOptionsPanel.equals(bugReportPanel.getParent())) {
+                    typeOptionsPanel.add(bugReportPanel);
+                    typeOptionsPanel.setPreferredSize(bugReportPanel.getPreferredSize());
+                    bugReportPanel.revalidate();
+                    bugReportPanel.repaint();
+                }
+                break;
+            case RFE:
+                descriptionLabel.setText(Bundle.getMessage("IssueReporter.descriptionLabel.feature"));
+                instructionsLabel.setText(Bundle.getMessage("IssueReporter.instructionsLabel.feature"));
+                typeOptionsPanel.remove(bugReportPanel);
+                break;
+            default:
+                log.error("Unexpected selected index {} for issue type", typeCB.getSelectedIndex(), new IllegalArgumentException());
+        }
+    }
+
+    private void submitBtnActionListener(ActionEvent e) {
+        IssueReport report = null;
+        switch (typeCB.getSelectedIndex()) {
+            case BUG:
+                report = new BugReport(titleText.getText(), bodyTA.getText(), profileCB.isSelected(), sysInfoCB.isSelected(), logsCB.isSelected());
+                break;
+            case RFE:
+                report = new EnhancementRequest(titleText.getText(), bodyTA.getText());
+                break;
+            default:
+                log.error("Unexpected selected index {} for issue type", typeCB.getSelectedIndex(), new IllegalArgumentException());
+        }
+        if (report != null) {
+            submitReport(report);
+        }
+    }
+
+    // package private
+    private void submitReport(IssueReport report) {
+        try {
+            URI uri = report.submit(repoCB.getItemAt(repoCB.getSelectedIndex()));
+            List<File> attachments = report.getAttachments();
+            if (!attachments.isEmpty()) {
+                JOptionPane.showMessageDialog(this,
+                        Bundle.getMessage("IssueReporter.attachments.message"),
+                        Bundle.getMessage("IssueReporter.attachments.title"),
+                        JOptionPane.INFORMATION_MESSAGE);
+                Desktop.getDesktop().open(attachments.get(0).getParentFile());
+            }
+            Desktop.getDesktop().browse(uri);
```

Comment:
This line throws an UnsupportedOperationException on Debian Linux for me.
`UnsupportedOperationException - if the current platform does not support the Desktop.Action.BROWSE action`

This seems to be a problem with Java unable to open the browser, so I don't know if it can be fixed, but the exception needs to be handled.

@pabender Can you test this on your Linux computer?

Reply:
I'm not sure, but it seems that [Desktop.isSupported](https://docs.oracle.com/javase/7/docs/api/java/awt/Desktop.html#isSupported(java.awt.Desktop.Action))( [Desktop.Action.BROWSE](https://docs.oracle.com/javase/7/docs/api/java/awt/Desktop.Action.html#BROWSE) ) can tell if web browsing is available.

Reply:
Thanks! 

We can check/catch that. What's the best way to handle it?  Open a Dialog box with the URL for somebody to manually open it?

Any idea why it's not working? Perhaps no default browser set?

Reply:
> We can check/catch that. What's the best way to handle it? Open a Dialog box with the URL for somebody to manually open it?

Yes, that seems to be a good solution. And to have a button "Copy link to clipboard".

Reply:
> Any idea why it's not working? Perhaps no default browser set?

[This page](https://docs.oracle.com/javase/tutorial/uiswing/misc/desktop.html) says:
On the Solaris Operating System and the Linux platform, this API is dependent on Gnome libraries. If those libraries are unavailable, this method will return false.

I have Gnome installed, but maybe not all Gnome. I use the Xfce desktop environment. I could try to install all Gnome to make it work, but on the other hand, it's useful that I have a computer there Java cannot open the browser so I can test that JMRI handles that.

Reply:
I think we need to come up with a different plan.  The URL is too long to really display:

![image](https://user-images.githubusercontent.com/2774117/89320018-ca11db00-d635-11ea-8207-97290cbd7e5e.png)

(It would be even longer if I'd put a lot of text in the boxes)

Reply:
I've been looking at a couple of other options that might help us which would get around the need for a GitHub account. But I am on a limited connection until early next week so might not be able to do too much until then.

Reply:
> I think we need to come up with a different plan. The URL is too long to really display:

I don't think you need to show the URL. It would probably be enough to have a button "Copy to clipboard".

https://stackoverflow.com/questions/3591945/copying-to-the-clipboard-in-java

Reply:
I've push a copy-to-clipboard approach just now.  it asks first, just in case the clipboard contents shouldn't be changed.

Reply:
> I've push a copy-to-clipboard approach just now. it asks first, just in case the clipboard contents shouldn't be changed.

Tested on Debian Linux and it works fine. The only thing I'm thinking of is that the "Copy to clipboard" dialog has the button "Skip" which I find somewhat confusing, since almost always that button is "Cancel". And I think it would be better to only close that dialog box and not the "Report Issue" dialog box when clicking on Skip/Cancel. The reason is that when that dialog comes up, the user may be confused and click on Skip/Cancel without really understanding what he is doing, and with the current solution, everything he has written in the "Report Issue" dialog is lost.

If the button is Cancel and only closes the "Copy to clipboard" dialog, the user will come back to the "Report Issue" dialog and can try again. I have been in this situation many times, confused with dialog boxes that I don't really understand.

But except this small comment, it now works fine on Debian Linux.